Form field lengths

mparke opened the pull request:
closes #1024 

Addresses default form field widths, and makes adjustments to heights and line-heights.

For differences in checkboxes, see: #1088 
For differences in selects, see #1130 

Includes:
- Default form field width + some additional width classes
- bumps form field height to `32px`
- removed `line-height` definitions where possible
- replaced `label-col` with bootstrap's `col-form-label` where possible
- updates label color
- introduced `row-label` for non-form labels
- fixed `line-height`/`padding` on buttons and combos where I broke them
- `font-sm` changed to `13px`, `font-xs` to `12px`
- updates styleguide forms section ( see screenshots )

there's a lot of small line changes where I've replaced:
```
<div class="label-col">
  <label>
</div>
```

with

```
<label class="col-form-label"></label>"
```

Screenshots:

Before:
![screen shot 2017-02-03 at 5 42 34 pm](https://cloud.githubusercontent.com/assets/709951/22611446/9a45bfe6-ea38-11e6-94df-37db9b622728.png)

After:
![screen shot 2017-02-03 at 5 42 27 pm](https://cloud.githubusercontent.com/assets/709951/22611450/a074a47c-ea38-11e6-865c-c8ed2f9176dd.png)

Before:
![screen shot 2017-02-03 at 5 42 48 pm](https://cloud.githubusercontent.com/assets/709951/22611457/abbca014-ea38-11e6-97aa-a8bfa3187cf6.png)

After: 
![screen shot 2017-02-03 at 5 42 44 pm](https://cloud.githubusercontent.com/assets/709951/22611461/b4ec2b00-ea38-11e6-97fb-20f014147d62.png)

Before:
![screen shot 2017-02-03 at 5 43 02 pm](https://cloud.githubusercontent.com/assets/709951/22611467/bc638518-ea38-11e6-82b1-b44db1f8185f.png)

After:
![screen shot 2017-02-03 at 5 42 54 pm](https://cloud.githubusercontent.com/assets/709951/22611472/c1d048ba-ea38-11e6-9bf1-41943ddc493d.png)

mparke commented on scss/components/form/form.scss:
```diff
@@ -4,37 +4,54 @@ form {
   }
 }
 
-// TOOD: remove input-container
```
default label and field width/height defined in here

mparke commented on scss/defaults.scss:
```diff
@@ -7,10 +7,10 @@ html {
   @include font-size(1.5);
```
font changes

mparke commented on scss/variables.scss:
```diff
@@ -43,8 +43,8 @@ $sidebar-width: 220px;
 
 /* BEGIN generic definitions */
 
-$input-height: 30px;
```
default height and padding

mparke commented on scss/components/form/form.scss:
```diff
@@ -4,37 +4,54 @@ form {
+.col-form-label {
+  color: $light-black; // $lightest-black
+  text-align: right;
+  line-height: 1.38;
```
I'm going to try a length unit here, there's a slight offset with field border last time I checked.

eatonphil commented on scss/components/Header.scss:
```diff
@@ -60,17 +63,18 @@ $header-padding: $grid-gutter-width-base / 2;
-        margin: 0 $header-padding 0 0;
+        padding: 14px 10px;
+
+        input.form-control {
```
Isn't this a duplicate of line 74?

eatonphil commented on scss/components/PrimaryPage.scss:
```diff
@@ -12,4 +12,43 @@
+
+    .btn {
+        border-radius: 0;
+    }
```
Why only define these here and not globally?

eatonphil commented on scss/components/PrimaryPage.scss:
```diff
@@ -12,4 +12,43 @@
+            height: $form-field-height;
+            float: left;
+        }
+    }
```
Ok sorry, why not define all these globally? I don't think thy should be here.

eatonphil commented on scss/components/PrimaryTable.scss:
```diff
@@ -36,4 +36,21 @@
+
+    .distro-style img {
+        margin-right: 15px;
+    }
```
:( Really would rather these specific rules not be in this PrimaryTable file.

eatonphil commented on scss/components/form/CheckboxInputCombo.scss:
```diff
@@ -1,16 +1,10 @@
-    .form-control {
-        display: inline;
-        width: auto;
-    }
```
What existing file uses CheckboxInputCombo and does dropping these display rules affect how errors are displayed?

eatonphil replied:
Actually it's not the form-control lines that are worrying so much as the CheckboxInputCombo display:block (now).

eatonphil commented on scss/components/form/RadioInputCombo.scss:
```diff
@@ -7,15 +7,9 @@
-    .form-control {
-        display: inline;
-        width: auto;
-    }
```
Same question here as the CheckboxInputCombo and error messages

eatonphil commented on scss/components/form/RadioSelectCombo.scss:
```diff
@@ -7,15 +7,9 @@
     .Radio,
     .Select {
         display: inline-block;
-        vertical-align: top;
```
Yeah?

mparke replied:
I think so. Let me double check the groups + group errors.

mparke replied:
Okay, I did find an issue here. Addressed with this change: https://github.com/linode/manager/pull/1135/commits/a6f765e2219a06af4846fba5582e84452f2ea759

making labels next to checks/radios vertical align middle now looks correct

eatonphil commented on scss/components/form/form.scss:
```diff
@@ -4,37 +4,53 @@ form {
+
+input.form-control.input-lg {
+  width: 325px;
+}
```
Why restrict these to `input`s only?

eatonphil commented on scss/defaults.scss:
```diff
@@ -26,6 +26,9 @@ body {
   color: $black;
+  font-weight: normal;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
```
Why is this only a font-medium rule?

mparke replied:
not sure what you mean, this is defined on `body`, you may be looking at collapsed lines in github?

eatonphil replied:
Oops. Sorry about that, you're right.

eatonphil replied:
What's with grayscale?

mparke replied:
it's the equivalent font-smoothing defined in `FF`, as far as I understand. however, let me back these effects out into another branch, we should compare separately

eatonphil commented on scss/defaults.scss:
```diff
@@ -85,3 +88,7 @@ hr {
 }
+
+strong {
+  font-weight: 600; // toned down from 'bolder'
```
500? pretty please?

eatonphil commented on scss/utilities.scss:
```diff
@@ -15,6 +15,9 @@
 // TODO: prettify
 .alert {
+  border-radius: 0;
+  height: 32px;  // for now, match input height
```
These can / are easily be multiline, can we set line-height instead?

mparke replied:
multiline with newlines? if not, i'd expect multiple errors to be separate strings, and subsequently multiple divs/spans/etc.

What would the goal of line height be in that case?

eatonphil replied:
If a form has multiple errors related to the entire form, we don't show a new alert box for each one. Right now we show a list (`ul`) of all the errors.

mparke replied:
Gotcha, let me find that, remove the height, and push a sample into the styleguide index

eatonphil commented on src/components/form/Radio.js:
```diff
@@ -2,24 +2,24 @@ import React, { PropTypes } from 'react';
-        {props.label ? <span className="Radio-label">{props.label}</span> : null}
-      </label>
-    </div>
+    <label className="col-form-label Radio">
```
Can .Radio and .Radio-label just extend these instead?

mparke replied:
i was on the fence between implicit and explicit declarations (scss @extend vs additional class) in cases like this. let me look again.

Have you started to favor @extend ?

eatonphil replied:
I'm also on the fence here. I can't argue strongly for either way, so might as well just keep it how you've got it I guess.

eatonphil commented on src/linodes/components/Linode.js:
```diff
@@ -207,10 +207,11 @@ function renderRow(props) {
         {renderBackupStatus(linode)}
       </td>
-      <td>
+      <td className="PrimaryTable-status-cell">
```
I'd prefer a less-specific rule like just right-aligning the last column or calling this PrimaryTable-rowOptions or literally just text-xs-right this td here.

eatonphil commented on src/linodes/linode/layouts/DashboardPage.js:
```diff
@@ -246,12 +244,10 @@ export class DashboardPage extends Component {
+              <label className="col-sm-4 col-form-label" htmlFor="lish-input">
+                Text console
+              </label>
+              <div className="col-sm-8  right">
```
You can drop all these `right` classes too (and `left` if they exist).

eatonphil commented:
Dropped all `.input-container`? Woot!

mparke commented:
Updated combo right-label alignment and added examples to index:

![screen shot 2017-02-06 at 3 27 31 pm](https://cloud.githubusercontent.com/assets/709951/22665008/17e220c6-ec81-11e6-8e59-6a5f44ec9246.png)

mparke commented:
Added multiple validation errors examples to index:

![screen shot 2017-02-06 at 3 27 20 pm](https://cloud.githubusercontent.com/assets/709951/22665018/2650b488-ec81-11e6-9dca-ea94cf5bfdad.png)
